src/stores/courses.ts

```diff
@@ -9,20 +9,41 @@ export interface CourseState {
 	courses: Array<Course>;
 }
 
+type CourseInfo =
+	{ course_id: number; course_name?: never } |
+	{ course_name: string; course_id?: never };
+
 export const useCourseStore = defineStore('courses', {
 	state: (): CourseState => ({
 		courses: []
 	}),
 	getters: {
-		getCourseByName: (state) => {
-			return (courseName: string) => state.courses.find((course) => course.course_name === courseName);
+		/**
+		 * returns a course in the store.
+		 * @param {CourseInfo} course_info - either a course_id or course_name
+		 * @returns {Course} the requested course
+		 */
```

Comment:
This is invalid.  This function takes the state parameter and returns a function that has the course_info parameter of type CourseInfo and returns a Course object.  It is not a function that takes a course_info parameter of type CourseInfo and returns a Course object.

Reply:
I agree that this returns a function, however as a getter, it is called like:
```
course_store.findCourse({course_id: 3 })
```
for example, which is why I wrote the documentation that way.

Reply:
However, it is invalid for jsDoc.  It has to actually be correct for what it actually is.

Reply:
If jsDoc is to be used, then this needs to be fixed.  This fails to generate valid documentation here.  Instead the tsserver complains about a @param tag with name course_info and no parameter with that name.  The @return type should be removed also.  That could be a real problem as typescript might try to use that.

This also occurs with some of the jsdoc in set_problems.ts and users.ts.